src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -38805,6 +38805,191 @@ static bool G(out object o)
+                //         (string f1, (C<string> f2, string f3)) = (null, (c, null)!);
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "c").WithArguments("C<string?>", "C<string>").WithLocation(16, 58),
+                // (16,61): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         (string f1, (C<string> f2, string f3)) = (null, (c, null)!);
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(16, 61),
+
+                // line 4
+                // (19,59): warning CS8619: Nullability of reference types in value of type 'C<string?>' doesn't match target type 'C<string>'.
+                //         (string g1, (C<string> g2, string g3)) = (null!, (c, null)!);
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "c").WithArguments("C<string?>", "C<string>").WithLocation(19, 59),
+                // (19,62): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         (string g1, (C<string> g2, string g3)) = (null!, (c, null)!);
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(19, 62)
+                );
+        }
+
+        [Fact, WorkItem(31370, "https://github.com/dotnet/roslyn/issues/31370")]
+        public void SuppressNullableWarning_Tuple()
+        {
+            // TODO2 crash
+            var source = @"
+class C<T> { }
+class C2
+{
+    static T Id<T>(T t) => t;
+    void M(C<string?> c)
+    {
+        // line 1
+        (string, (C<string>, string)) t1 = (null, (c, null));
+        t1.Item1.ToString(); // warn
+        Id(t1).Item1.ToString(); // no warn
+
+        // line 2
+        (string, (C<string>, string)) t2 = (null, (c, null))!;
+        t2.Item1.ToString(); // warn
+        Id(t2).Item1.ToString(); // no warn
+
+        // line 3
+        (string, (C<string>, string)) t3 = (null, (c, null)!);
+        t3.Item1.ToString(); // warn
+        Id(t3).Item1.ToString(); // no warn
+
+        // line 4
+        (string, (C<string>, string)) t4 = (null!, (c, null)!);
+        t4.Item1.ToString(); // no warn
+        Id(t4).Item1.ToString(); // no warn
+
+        // line 5
+        (string, (C<string>, string)) t5 = (null!, (c!, null!)); // no warn
+        t5.Item1.ToString(); // no warn
+        Id(t5).Item1.ToString(); // no warn
+
+        // line 6
+        (string, (C<string>, string)) t6 = (null!, (c!, null!)!); // no warn
+        t6.Item1.ToString(); // no warn
+        Id(t6).Item1.ToString(); // no warn
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // line 1
+                // (9,44): warning CS8619: Nullability of reference types in value of type '(string?, (C<string>, string))' doesn't match target type '(string, (C<string>, string))'.
```

Comment:
Note that there are two warnings for the assignments here, but three on the deconstruction above. Is that intentional?

Reply:
Is this comment referring to something that was force-pushed out of existence?

Reply:
Yup